docs/src/docs/asciidoc/user/clustered-cache.adoc

```diff
@@ -9,77 +9,132 @@ ifdef::notBuildingForSite[]
 include::menu.adoc[]
 endif::notBuildingForSite[]
 
+:imagesdir: ../../../../images/design/clustered
+
 == Introduction
 
 To enable Clustering with Terracotta, firstly you will have to start the http://terracotta.org[Terracotta server] configured with clustered storage.
 And then need a cache manager with clustering capabilities so that the caches can consume the clustered storage.
 
+Detailed instructions on how to configure and start a Terracotta server array can be found elsewhere.
+In this document we will use the minimalist configuration and setup for the server that is relevant in the context of this document.
+
+[[start-server]]
 == Starting Terracotta server
 
+You can start the server with the following configuration.
+It has the bare minimum configuration required for the samples in the rest of the document to work.
+
+[source,xml]
+----
+include::{sourcedir}/clustered/client/src/test/resources/configs/docs/tc-config.xml[]
+----
+
+The above configuration defines two named _server off-heap resources_:
+
+<1> An off-heap resource of 128 MB size named `primary-server-resource`.
+<2> Another off-heap resource named `secondary-server-resource` with 96 MB capacity.
+
+The clients (cache managers) can consume storage from these named resources by configuring the cache manager accordingly.
+The rest of the document explains in detail how you can configure cache managers and caches to consume the server's off-heap resources.
+
 Assuming that you have the clustered EhCache kit available locally, start with extracting the *ehcache-clustered* kit.
-Change to your extracted directory and then execute the *start-tc-server* script as below to start the Terracotta server with default configuration:
+Change to your extracted directory and then execute the *start-tc-server* script as below to start the Terracotta server with the above configuration:
 
-On Windows: +
-Execute the *terracotta-5.x.y-SNAPSHOT/server/bin/start-tc-server.bat* script.
+On Windows:
+[source,cmd]
+----
+cd <path/to/terracotta/kit>/server/bin
+start-tc-server.bat -f <path/to/server/config>/tc-config.xml
+----
 
 On Unix/Mac:
 [source,bash]
 ----
 cd terracotta-5.x.y-SNAPSHOT/server/bin
-./start-tc-server.sh
+./start-tc-server.sh -f <path/to/server/config>/tc-config.xml
 ----
 
 NOTE: You will need to have `JAVA_HOME` set to JDK8 while starting the Terracotta server.
 
 Check for the below `INFO` log to confirm if the server started successfully,
 `Terracotta Server instance has started up as ACTIVE node on 0:0:0:0:0:0:0:0:9510 successfully, and is now ready for work.`
 
+== Clustering concepts
+
+In this section we discuss some Terracotta clustering terms and concepts that users need to understand before creating cache managers with clustering support.
+
+Clustered storage::
```

Comment:
I am wondering if we should not simply use _Ehcache entity_. Problem with _clustered storage manager_ is that it does not limit itself to Ehcache just by name.